### PR TITLE
Workaround for old pip versions that are not exposed as importable modules.

### PR DESCRIPTION
**Pull request by adferrand**
So, it appears, after last Certbot `test-everything` execution, that some supported distributions use a very old version of pip. And theses old versions do not expose pip as an importable module: so `python -m pip` call will not work.

To call `pip` while respecting the actual environment used to launch pip, this PR extract the actual pip executable path from `sys.executable`. Indeed in virtual environment, the packaged pip executable wrapper will execute effectively in the relevant virtual environment.

### CODE REVIEW

File: pipstrap.py [hunk 1]
```diff
@@ -150,7 +150,10 @@ def get_index_base():
 
 
 def main():
-    pip_version = StrictVersion(check_output([executable, '-m', 'pip', '--version'])
+    # We should do 'python -m pip' here, but some old pip versions are not exposed
+    # as an importable module for Python, but only as an executable wrapper.
+    pip_executable = join(dirname(executable), basename(executable).replace('python', 'pip'))
```
Comment (erikrose) [inline pipstrap.py]: Hmm, this makes a lot of assumptions:
1. That pip sits next to Python (which is a good bet in a venv but not necessarily elsewhere)
2. That Python has "python" in its name
3. That pip can be found by replacing "python" in the executable's name with "pip"

That strikes me as rather a lot. :-) Should we consider going back to the original strategy of just finding a thing called "pip" in the $PATH? We do set the PATH properly in certbot-auto before calling:

`PATH="$VENV_BIN:$PATH" "$VENV_BIN/python" "$TEMP_DIR/pipstrap.py"`

Comment (adferrand) [reply pipstrap.py]: See my final comment in the PR discussion.

**Comment by adferrand** [comment]
In fact, certbot-auto is not the problem here, but more _venv_common.py that needs to call pipstrap in the context of the virtual environment. It was not easy from a python script, on the contrary to a shell script. That's why I think that the PR for 2.0.0 was solving things elegantly, by not relying on some PATH manipulation. 

However I was not taking into account the crappy implementations of pip that was done several years ago, and for which Certbot still needs to maintain a compatibility. 

However, I have to admit that my attempt to solve that from pipstrap would weaken your code. Upstream should be altered in the last resort. 

And I found a way to solve that from Certbot, using the version 1.5.0 of pipstrap. 

So I will close this PR, and you will not have to  worry about that anymore.

**Comment by adferrand** [comment]
So about the version 2.0.0 of pipstrap, I still think that it is an improvement, but with less compatibility towards very old versions of pip.

In my mind it make sense to keep it, as a major version can break compatibility. And certbot may reuse it once these old compatibilities will be dropped, maybe with the new packaging system. 

Now it is your call ^^

**Comment by erikrose** [comment]
I think we did make it worse, because, after all, the point of pipstrap is to work on old installations. However, I don't think anyone but certbot is still using it, so I'm not going to invest further effort unless certbot needs another fix. Then I'll probably revert to the old behavior.

**Pull request closed without merging**